packages/types/src/Credential.ts

```diff
@@ -21,8 +21,14 @@ export interface ICredential {
   claim: IClaim
   claimNonceMap: Record<Hash, string>
   claimHashes: Hash[]
-  claimerSignature?: DidSignature & { challenge?: string }
   delegationId: IDelegationNode['id'] | null
   legitimations: ICredential[]
   rootHash: Hash
 }
+
+export interface ICredentialPresentation extends ICredential {
+  claimerSignature: DidSignature & { challenge?: string }
+}
+
+export type CredentialOrPresentation = ICredential &
```

Comment:
A normal type union doesn't work in this case?

Reply:
weirdly a union of the two does not result in `{...credentialProps, claimerSignature?}`. Don't ask me why